packages/admin-rte/package.json

```diff
@@ -7,6 +7,7 @@
         "lib"
     ],
     "devDependencies": {
+        "@comet/admin": "^2.0.0-alpha.8",
```

Comment:
das müsste ein peerDependency sein.

Und das ist wirklich gut zu überlegen ob wir das wollen, @rabengraph wird nicht begeistert sein :)

("nur" für die buttons)

Reply:
@nsams I tried it as peerDependency before, that caused the pipeline to fail: https://gitlab.vivid-planet.com/comet/comet-admin/-/jobs/286567 🤷🏻‍♂️ 

It's not just for the buttons, we also need FieldContainer (added in  https://github.com/vivid-planet/comet-admin/pull/461). And if we manually recreate the button-components we would need admin-icons as a dependency.

Why would we want the rte to work independently of comet-admin? We already require `@material-ui/core` and `@material-ui/icons`, does adding `@comet/admin` and `@comet/admin-icons` really make a meaningful difference in ability to use the package or in package-size? 

If we do want rte to work independently, we could maybe:
- create a "generic" link-form in the rte-package using standard mui-components and extend the RTE in the admin-package to add a new form, built using comet-admin components (rte will not fully match the comet-design by default)
- recreate the comet-admin-components inside the rte-package (requires maintaining duplicate code, risks inconsistencies in style/usability which is currently the case)

Reply:
There a few aspects:

### Efforts to keep rte independent from core

In version 1 rte has a peer dependency to comet/admin, it is new in the next branch, that rte is fully independent.
Would be nice to keep it like this, but its no must-have.
I prefer a peer dependency to @comet/admin-icons.

### Link Dialog

The Link/ToolbarButton with the Link Dialog (this is what this PR changes) needs to match the design of comet/admin, this is obvious. But in reality the  `/extension/Link/ToolbarButton.tsx` is NEVER used in a real comet/admin application, because this Link Component is only capable of creating text-links. 
In comet/cms we have a custom ToolbarButton-Component.

My conclusion is:

- To not seriously maintain this `/extension/Link/ToolbarButton.tsx`, because in fact this component is only used in applications where rte is used standalone without comet/admin.
- If we want a `/extension/Link/ToolbarButton.tsx` which matches comet/admin design for use-cases where comet/admin is not used within comet/cms (because in this case we have a more secific `/extension/Link/ToolbarButton.tsx` anyway ), then we could maintain another `/extension/Link/ToolbarButton.tsx` in comet/admin.

Reply:
@nsams @rabengraph I now removed the dependency and used only mui-components for the LinkDialog. 
I have also added props to override those components using the comet-theme.
What do you think of this solution?

Reply:
> What do you think of this solution?

complicated solution for a non-problem :)

I would simply use DefaultCancelButton as you copied it already and don't do complicated things using theme. We don't use `/extension/Link/ToolbarButton.tsx` it's just a demo thing!

Reply:
@nsams you're right, this is probably overkill if this will never actually be used in production.  
I will remove the theming stuff.

Reply:
>it's just a demo thing!

It is actually in use in production, in projects without comet/admin, where we just need a simple text-link.

Reply:
@rabengraph @nsams if we are using this in production the buttons should defiantly look like the other comet-buttons, when using the comet-theme. I know it adds complexity, but I think it makes sense to allow overriding of the buttons with the theme, like i added in 6a3adf5e716ba99d4429d71c9fd4066bd5264dcc and later removed.

Reply:
@jamesricky we re using it on some production sites, which do not even include comet/admin. In those cases it must not look like comet.